index.bs

```diff
@@ -0,0 +1,221 @@
+
+
+
+
+Processing model {#sec-processing-model}
+========================================
+
+
+Modifications to the HTML specification {#sec-modifications-HTML}
+--------------------------------------------------------
+
+<em>This section will be removed once the <a href=https://html.spec.whatwg.org/multipage>HTML specification</a> has been modified.</em>
+
+For each <a>element</a>, there is a:
+* <dfn>currentFrameStartingPoint</dfn> which is initially set to <code>null</code>.
+* <dfn>previousFrameStartingPoint</dfn> which is initially set to <code>null</code>.
+* <dfn>currentVisualRepresentation</dfn> which is initially set to <code>null</code>.
+* <dfn>previousVisualRepresentation</dfn> with is initially set to <code>null</code>.
+
+<div algorithm="additions to update rendering">
+    In the <a>update the rendering</a> step of the <a>event loop processing model</a>, add a step right after the step that calls <a>mark paint timing</a>:
+
+    1. Invoke the algorithm to <a>evaluate the layout instability</a> of a document with the complete set of <a>element</a>s for that {{Document}}.
+</div>
+
+
+Evaluate Layout Instability {#sec-eval-layout-instability}
+--------------------------------------------------------
+
+<div algorithm="evaluate the layout instability">
+    When asked to <dfn export>evaluate the layout instability</dfn>, with <var>elements</var> as the list of elements to be evaluated, run the following steps:
+
+    1. Let <var>unstable elements</var> be an empty list.
+    1. Let <var>total width</var> be the current <a>viewport</a> width.
+    1. Let <var>total height</var> be the current <a>viewport</a> height. 
+    1. For each <var>element</var> of <var>elements</var>:
+        1. Set <a>currentFrameStartingPoint</a> to the <a>starting point</a> of <var>element</var>.
+        1. Set <a>currentVisualRepresentation</a> to the <a>visual representation</a> of <var>element</var>.
+        1. Let <var>unstable</var> be the boolean returned from calling <a>identify an unstable element</a> with <var>element</var>’s <a>currentFrameStartingPoint</a> and <a>previousFrameStartingPoint</a>.  
+        1. If <var>unstable</var> is true, add a tuple to <var>unstable elements</var> with:
+            1. <a>currentVisualRepresentation</a> of <var>element</var>
+            1. <a>previousVisualRepresentation</a> of <var>element</var>
+            1. <var ignore=''>shiftFraction</var> of <var>element</var>:
+                1. Set to 0 if <a>previousFrameStartingPoint</a> is null.
+                1. Set to the max(abs(<a>previousFrameStartingPoint</a>.height - <a>currentFrameStartingPoint</a>.height) / <var>total height</var> , abs(<a>previousFrameStartingPoint</a>.width - <a>currentFrameStartingPoint</a>.width) / <var>total width</var>).
```

Comment:
These points also seem underspecified (what is their type?). And we should try to avoid using JavaScript notation (".height") and instead be more verbose: maybe "currentFrameStartingPoint's height", with the appropriate link to the height.

Reply:
what do you mean by link to the height?  If we have a link to this above (which I realized we don't yet) do we need to add it here?

Reply:
Well, yes. I'm not even sure you mean height here. For the height above, it is for a viewport which is a rectangle. You're talking about height of a point here.

Reply:
Well this is the max distance it has moved.  It can move in the horizontal or vertical direction, so we do want the height coordinate of the current point and the previous point to determine the distance it moved in the horizontal direction and same thing for the vertical direction. 

Should I state that differently here so it is more clear?

Reply:
It's a bit confusing to refer to the "height" of a point.  I'd suggest "y-coordinate" or "y-value" instead.

Reply:
Done.  And I incorporated the new denominator changes that we discussed with Bryan.  ie denominator = max (viewport height, viewport width)